motr/idx_dix.c

```diff
@@ -656,6 +660,10 @@ static void dixreq_completed_ast(struct m0_sm_group *grp, struct m0_sm_ast *ast)
 	oi->oi_ar.ar_ast.sa_cb = (rc == 0) ? idx_op_ast_complete :
 					     idx_op_ast_fail;
 	oi->oi_in_completion = true;
+	/* XXX: it looks like there is no need to set up an ast for that.
+	 * The groups are the same. We can just call the callback right here?
+	 *   oi->oi_ar.ar_ast.sa_cb(oi->oi_sm_grp, &oi->oi_ar.ar_ast)
```

Comment:
Should be okay, unless we are in a different group than the one in which the callback is supposed to be executed, else it may lead to inconsistencies.